### PR TITLE
Issue/4416: Updates account details / avatar on login

### PR DESCRIPTION
**Pull request by frosty**
Refs #4416 and #4613. `AccountService` now sends a notification when the user's account details are updated, so that the Me UI can update itself. I've also put in some changes so that local `AccountSettings` changes update the current `WPAccount` and send out the same notification. 

I'm not a big fan of the fact that my `WPAccount` extension sends out the same notification as `AccountService`. I'm not sure if some sort of observer pattern might make more sense here, to watch out for changes to the account object.

Work in progress!

### CODE REVIEW

File: WordPress/Classes/ViewRelated/MeViewController.swift [hunk 1]
```diff
@@ -300,6 +303,9 @@ class MeViewController: UITableViewController, UIViewControllerRestoration {
     // MARK: - Notification observers
 
     func refreshModelWithNotification(notification: NSNotification) {
+        // Regenerate the header view if the account has changed
+        cachedHeaderView = MeHeaderView()
```
Comment (koke) [inline WordPress/Classes/ViewRelated/MeViewController.swift]: I'm not sure why is this needed. `reloadViewModel` should already update the existing header view

Comment (frosty) [inline WordPress/Classes/ViewRelated/MeViewController.swift]: I think I was seeing an issue when logging out and back in where the previous account details would be displayed until the new ones had been loaded. This line means that on logout the header view is instantly replaced with the generic account icon / name. I can check this though!

**Comment by koke** [comment]
Taking care of this in #4660
